java/cdk-environment/src/main/java/sleeper/environment/cdk/buildec2/BuildEC2Stack.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2022 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package sleeper.environment.cdk.buildec2;
+
+import sleeper.environment.cdk.config.AppContext;
+import sleeper.environment.cdk.util.MyIpUtil;
+import software.amazon.awscdk.CfnOutput;
+import software.amazon.awscdk.Stack;
+import software.amazon.awscdk.StackProps;
+import software.amazon.awscdk.services.ec2.*;
+import software.constructs.Construct;
+
+import java.security.KeyPair;
+import java.util.Collections;
+import java.util.UUID;
+
+import static sleeper.environment.cdk.config.AppParameters.VPC;
+
+public class BuildEC2Stack extends Stack {
+
+    public BuildEC2Stack(Construct scope, StackProps props, IVpc inheritVpc) {
+        super(scope, props.getStackName(), props);
+        AppContext context = AppContext.of(this);
+        BuildEC2Parameters params = BuildEC2Parameters.from(context);
+        IVpc vpc = context.getOrDefault(VPC, this, inheritVpc);
+
+        SecurityGroup allowSsh = SecurityGroup.Builder.create(this, "AllowSsh")
+                .vpc(vpc)
+                .description("Allow SSH inbound traffic")
+                .allowAllOutbound(true)
+                .build();
+        allowSsh.addIngressRule(Peer.ipv4(MyIpUtil.findMyIp() + "/32"), Port.tcp(22));
```

Comment:
This is quite a big method, wonder whether putting the above SecurityGroup creation in a private method would help.

Reply:
Done.